tests/array_split.py

```diff
@@ -22,127 +22,110 @@
 import cunumeric as num
 
 
-def test(min_size, max_size):
-    if min_size >= max_size:
-        max_size = min_size + 1
+def test(dim):
     # Seed the random generator with a random number
     np.random.seed(416)
     print("test split")
     test_routine = ["array_split"]
-    # test the split routines on 3D array [1:10, 1:10, 1:10] w/ integers,
-    # list of indicies. vsplit, hsplit, dsplit are included
+    # test the split routines on empty, singleton, 2D and 3D arrays
+    # w/ integers, list of indicies. vsplit, hsplit, dsplit are included
     # in the following loops(axis = 0: vsplit, 1: hsplit, 2: dsplit)
-    for i in range(min_size, max_size):
-        for j in range(min_size, max_size):
-            for k in range(min_size, max_size):
-                a = np.random.randn(i, j, k)
-                for axis in range(a.ndim):
-                    input_arr = []
-                    even_div = None  # a.shape[axis]
-                    uneven_div = None  # a.shape[axis]
-                    for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
-                        if a.shape[axis] % div == 0:
-                            even_div = div
-                        else:
-                            uneven_div = div
-                        if even_div is not None and uneven_div is not None:
-                            break
-                    if even_div is None:
-                        even_div = a.shape[axis]
-                    # divisible integer
-                    input_arr.append(even_div)
-                    # indivisble integer
-                    if even_div != uneven_div and uneven_div is not None:
-                        input_arr.append(uneven_div)
-                    # integer larger than shape[axis]
-                    input_arr.append(a.shape[axis] + np.random.randint(1, 10))
-                    # indices array which has points
-                    # within the target dimension of the src array
-                    if a.shape[axis] > 1:
-                        input_arr.append(
-                            list(range(1, a.shape[axis], even_div))
-                        )
-                    # indices array which has points
-                    # out of the target dimension of the src array
-                    input_arr.append(
-                        list(
-                            range(
-                                0,
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                even_div,
-                            )
-                        )
+
+    input_shapes = [(0), (1), (1, 1), (1, 1, 1), (dim, dim), (dim, dim, dim)]
```

Comment:
I guess you could also add empty arrays with more convoluted shapes, like (0, 10).